cyto_dl/nn/track_sequence_predictor.py

```diff
@@ -0,0 +1,78 @@
+import math
+
+import torch
+from einops import rearrange
+from monai.networks.nets import Regressor
+from timm.models.vision_transformer import Block
+
+
+class TrackClassifier(torch.nn.Module):
+    """Transformer that encodes images in a sequence as tokens and classifies each of them."""
+
+    def __init__(
+        self,
+        patch_size,
+        pos_embedding_length=120,
+        emb_dim=128,
+        num_layer=8,
+        num_head=4,
+        num_classes=2,
+    ) -> None:
+        super().__init__()
+
+        self.register_buffer("pos_embedding", positionalencoding1d(emb_dim, pos_embedding_length))
+        self.pos_embedding.requires_grad = False
+
+        self.image_encoder = Regressor(
+            in_shape=patch_size, out_shape=emb_dim, channels=[8, 16, 32], strides=[2, 2, 2]
+        )
+
+        self.transformer = torch.nn.Sequential(
+            *[Block(emb_dim, num_head) for _ in range(num_layer)]
+        )
+
+        self.layer_norm = torch.nn.LayerNorm(emb_dim)
+        self.classifier = torch.nn.Linear(emb_dim, num_classes)
+
+    def forward(self, img):
+        img = img.as_tensor()
+        # embed images individually
+        img = rearrange(img, "b c h w -> c b h w")
+        # return  t b c
+        patches = self.image_encoder(img).unsqueeze(1)
```

Comment:
does this output patches? this image encoder returns an embedding of size `embed_dim` from what I can understand?

Reply:
also what is t in t b c